### DISCUSSION TITLE
CPU and Memory level metrics

### DISCUSSION THREAD

**Discussion by Khushbukela** [prometheus/client_golang#954] [Q&A]
I have golang based rest API. 
Prometheus shows golang based metrics but does not show CPU and memory level metrics. 
How should I export memory and CPU level metrics. 
when I specify `process_cpu_seconds_total` as name for counter then it will panic , because it already in use. 
need help!

**Comment by Khushbukela** [comment]
Currently, working on mac? may be this is the issue.

**Reply by fastcat** [reply]
I suspect macOS is your issue, yes: `process_cpu_seconds_total` comes from this collector: https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/collectors#NewProcessCollector

Whose documentation notes:
> The collector only works on operating systems with a Linux-style proc filesystem and on Microsoft Windows. On other operating systems, it will not collect any metrics.

macOS has no `/proc`
